CleverPush/Source/CPAppBannerModuleInstance.m

```diff
@@ -196,7 +196,15 @@ - (void)getBanners:(NSString*)channelId bannerId:(NSString*)bannerId notificatio
 
     NSMutableURLRequest* request = [[CleverPushHTTPClient sharedClient] requestWithMethod:HTTP_GET path:bannersPath];
     [CleverPush enqueueRequest:request onSuccess:^(NSDictionary* result) {
-        NSArray *jsonBanners = [result objectForKey:@"banners"];
+        NSMutableArray *jsonBanners = [[NSMutableArray alloc] init];
+        if (groupId != nil && ![groupId isEqualToString:@""]) {
+            NSString *expression=[NSString stringWithFormat:@"SELF contains '%@'",groupId];
```

Comment:
Code style (spaces)

Also why do we need this expression? Can't we directly check for banners which have the field `group`= groupId?

Reply:
Hello @gbhrdt  We are getting the array of **banners**, and from that, we need to filter which array element contains the groupId. That's why we need to do the filter.
Before using filter, I have used a loop for adding the objects with the same **groupId**, but due to less code, I have used an expression for getting the filtered data.

Reply:
Hello @gbhrdt  We are getting the array of **banners**, and from that, we need to filter which array element contains the groupId. That's why we need to do the filter.
Before using filter, I have used a loop for adding the objects with the same **groupId**, but due to less code, I have used an expression for getting the filtered data.